app/scripts/modules/core/src/pipeline/config/stages/findArtifactFromExecution/findArtifactFromExecutionConfig.html

```diff
@@ -0,0 +1,52 @@
+<div class="form-horizontal clearfix">
+  <h4>Pipeline Selector</h4>
+  <stage-config-field label="Application">
+    <ui-select
+        class="form-control input-sm"
+        on-select="ctrl.onApplicationSelect()"
+        ng-model="ctrl.stage.application"
+        ng-if="ctrl.state.applicationsLoaded">
+      <ui-select-match placeholder="None">{{$select.selected}}</ui-select-match>
+      <ui-select-choices
+          repeat="application in ctrl.state.applications | filter: $select.search | limitTo: ctrl.state.infiniteScroll.currentItems"
+          infinite-scroll="ctrl.addMoreApplications()"
+          infinite-scroll-distance="2" >
+        <div ng-bind-html="application | highlight: $select.search"></div>
+      </ui-select-choices>
+    </ui-select>
+  </stage-config-field>
+  <stage-config-field label="Pipeline">
+    <div ng-if="ctrl.stage.application && ctrl.state.pipelinesLoaded">
```

Comment:
Ideally we would use spinners here rather than having the field just pop in. 

Also, a little unusual for the field to be hidden while application hasn't been selected - I would just have an empty dropdown.

Reply:
The pipelines seem to load so fast the spinner (which i'm guessing should only be present when the drop down is open, and a new application was just selected) won't even get seen.